lib/cell.rb

```diff
@@ -1,13 +1,14 @@
 class Cell
-  attr_reader :coordinate, :ship
+  attr_reader :coordinate, :ship, :fired_upon
 
-  def initialize(coordinate_parameter)
+  def initialize(coordinate_parameter, ship = nil)
     @coordinate = coordinate_parameter
-    @ship = nil
+    @ship = ship
+    @empty = true
   end
 
   def empty?
-    @ship == nil
+    @empty
```

Comment:
(We talked about this in person, but I'll summarize for the record): This method could work, if we added a line to the place_ship method to change @empty to false. My preference is for the way we had it, because it's a little cleaner and requires one less variable. If we do go back to the original, we'll need to update the fire_upon method -- see comment below.